tools/diagnostics/nsdiff

```diff
@@ -0,0 +1,673 @@
+                                        data['view']))
+                    else:
+                        copyfile(join(abspath(dirName), fname), join(sout,
+                            fname))
+
+        # Check for a log file
+        (dirname, filename, name, extension) = self.split_name(sin)
+        log_filename = "{}.log".format(filename)
+        log_filepath = "{}/{}".format(dirname, log_filename)
+        if (os.path.exists(log_filepath)):
+            # Extract (view no, ppSeqNo) tuple from log file(s)
+            # Last "batch ordered"
+            process_grep = subprocess.Popen([ "grep", "-e"
+                "{}:.* ordered batch request, view no".format(
+                filename), log_filepath],
+                stdout=subprocess.PIPE, shell=False)
+            process_tail = subprocess.Popen([ 'tail', '-1' ],
+                stdin=process_grep.stdout, stdout=subprocess.PIPE,
+                shell=False)
+            process_cut1 = subprocess.Popen([ 'cut', '-f', '5',
+                '-d', '|' ], stdin=process_tail.stdout,
+                stdout=subprocess.PIPE, shell=False)
+            process_cut2 = subprocess.Popen([ 'cut', '-d', ',',
+                '-f', '2,3' ], stdin=process_cut1.stdout,
+                stdout=subprocess.PIPE, shell=False)
+            process_grep.stdout.close()
+            process_tail.stdout.close()
+            process_cut1.stdout.close()
+
+            batch_ordered_tuple = process_cut2.communicate()[0]
+
+            # Last "batch executed"
+            process_grep = subprocess.Popen([ "grep", "-e"
+                "{} executing Ordered batch".format(
+                filename), log_filepath],
+                stdout=subprocess.PIPE, shell=False)
+            process_tail = subprocess.Popen([ 'tail', '-1' ],
+                stdin=process_grep.stdout, stdout=subprocess.PIPE,
+                shell=False)
+            process_cut1 = subprocess.Popen([ 'cut', '-f', '5',
+                '-d', '|' ], stdin=process_tail.stdout,
+                stdout=subprocess.PIPE, shell=False)
+            process_cut2 = subprocess.Popen([ 'cut', '-d', ' ',
+                '-f', '6,7' ], stdin=process_cut1.stdout,
+                stdout=subprocess.PIPE, shell=False)
+            process_grep.stdout.close()
+            process_tail.stdout.close()
+            process_cut1.stdout.close()
+
+            batch_executed_tuple = process_cut2.communicate()[0]
+            # Use the filename of "log" so that the *nix-diff can be done on a
+            # single directory.
+            out_filename = join(sout, "log")
+            with open(out_filename, "a") as file_handle:
+                file_handle.write("last batch ordered: {}".format(
+                    batch_ordered_tuple.decode()))
+                file_handle.write("last batch executed: {}".format(
+                    batch_executed_tuple.decode()))
+
+    def diff(self, s1, s2):
+        """Perform a recursive *nix-diff on two directories"""
```

Comment:
Are we sure that `diff` gives equal results for the same data (same txns) written at different time? Is there any db-specific metadata there?